util/src/hashdb.rs

```diff
@@ -42,7 +95,29 @@ pub trait HashDB: AsHashDB + Send + Sync {
 	///   assert_eq!(m.get(&hash).unwrap(), hello_bytes);
 	/// }
 	/// ```
-	fn get(&self, key: &H256) -> Option<DBValue>;
+	fn get(&self, key: &H256) -> Option<DBValue> {
+		let mut o_func = Some(DBValue::from_slice);
+		let mut output = None;
+
+		{
+			let mut wrapper = |key: &[u8]| { output = Some((o_func.take().unwrap())(key)); };
```

Comment:
could be simplified just to `DBValue::from_slice(key)`

codebase style replaces unwrap with `expect("qualify why it never will panic; qed")`